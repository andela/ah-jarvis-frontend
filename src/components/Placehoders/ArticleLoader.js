import React from 'react';
import ContentLoader from 'react-content-loader';


const ArticleLoader = props => (
  <ContentLoader
    height={160}
    width={400}
    speed={1.5}
    primaryColor="#dfdfdf"
    secondaryColor="#ffffff"
    {...props}
  >
    <circle cx="13.42635" cy="14.42635" r="10.42635" />
    <rect x="30" y="7.05" rx="4" ry="4" width="87" height="5.09" />
    <rect x="29" y="16.05" rx="4" ry="4" width="50" height="3.88" />
    <rect x="-1" y="106.05" rx="0" ry="0" width="262.21680000000003" height="9.2682" />
    <circle cx="204" cy="45.05" r="1" />
    <rect x="1" y="29.05" rx="0" ry="0" width="381.76" height="65.45" />
    <rect x="4" y="143.05" rx="0" ry="0" width="0" height="0" />
    <rect x="-6" y="121.88" rx="0" ry="0" width="384.52" height="4.219499999999999" />
    <rect x="1" y="133.81" rx="0" ry="0" width="36.75" height="4.2" />
    <circle cx="5.92" cy="145.97" r="4.2" />
    <circle cx="372.27" cy="141.97" r="4.2" />
  </ContentLoader>

);


export default ArticleLoader;
