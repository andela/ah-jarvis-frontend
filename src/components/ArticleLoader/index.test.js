import { shallow } from 'enzyme';
import React from 'react';
import ArticleLoader from '.';

describe('<ArticleLoader />', () => {
  it('renders  <ArticleLoader /> components', () => {
    const snap = shallow(<ArticleLoader />);
    expect(snap).toMatchSnapshot();
  });
});
