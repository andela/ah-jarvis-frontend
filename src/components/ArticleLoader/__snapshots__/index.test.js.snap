// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleLoader /> renders  <ArticleLoader /> components 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="preloader-wrapper big active"
>
        <div
                className="spinner-layer spinner-blue-only"
        >
                <div
                        className="circle-clipper left"
                >
                        <div
                                className="circle"
                        />
                </div>
                <div
                        className="gap-patch"
                >
                        <div
                                className="circle"
                        />
                </div>
                <div
                        className="circle-clipper right"
                >
                        <div
                                className="circle"
                        />
                </div>
        </div>
</div>,
      "className": "flex-center",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="spinner-layer spinner-blue-only"
>
          <div
                    className="circle-clipper left"
          >
                    <div
                              className="circle"
                    />
          </div>
          <div
                    className="gap-patch"
          >
                    <div
                              className="circle"
                    />
          </div>
          <div
                    className="circle-clipper right"
          >
                    <div
                              className="circle"
                    />
          </div>
</div>,
        "className": "preloader-wrapper big active",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="circle-clipper left"
>
              <div
                            className="circle"
              />
</div>,
            <div
              className="gap-patch"
>
              <div
                            className="circle"
              />
</div>,
            <div
              className="circle-clipper right"
>
              <div
                            className="circle"
              />
</div>,
          ],
          "className": "spinner-layer spinner-blue-only",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="circle"
/>,
              "className": "circle-clipper left",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "circle",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="circle"
/>,
              "className": "gap-patch",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "circle",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="circle"
/>,
              "className": "circle-clipper right",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "circle",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="preloader-wrapper big active"
>
          <div
                    className="spinner-layer spinner-blue-only"
          >
                    <div
                              className="circle-clipper left"
                    >
                              <div
                                        className="circle"
                              />
                    </div>
                    <div
                              className="gap-patch"
                    >
                              <div
                                        className="circle"
                              />
                    </div>
                    <div
                              className="circle-clipper right"
                    >
                              <div
                                        className="circle"
                              />
                    </div>
          </div>
</div>,
        "className": "flex-center",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="spinner-layer spinner-blue-only"
>
            <div
                        className="circle-clipper left"
            >
                        <div
                                    className="circle"
                        />
            </div>
            <div
                        className="gap-patch"
            >
                        <div
                                    className="circle"
                        />
            </div>
            <div
                        className="circle-clipper right"
            >
                        <div
                                    className="circle"
                        />
            </div>
</div>,
          "className": "preloader-wrapper big active",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="circle-clipper left"
>
                <div
                                className="circle"
                />
</div>,
              <div
                className="gap-patch"
>
                <div
                                className="circle"
                />
</div>,
              <div
                className="circle-clipper right"
>
                <div
                                className="circle"
                />
</div>,
            ],
            "className": "spinner-layer spinner-blue-only",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="circle"
/>,
                "className": "circle-clipper left",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "circle",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="circle"
/>,
                "className": "gap-patch",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "circle",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="circle"
/>,
                "className": "circle-clipper right",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "circle",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
